test/unit/SourceLocator/Type/AutoloadSourceLocatorTest.php

```diff
@@ -302,4 +308,77 @@ public function autoload(string $className) : bool
 
         return true;
     }
+
+    /**
+     * @runInSeparateProcess
+     */
+    public function testPharWorks() : void
+    {
+        require_once 'phar://' . __DIR__ . '/../../Fixture/autoload.phar/vendor/autoload.php';
+        spl_autoload_register(static function (string $class) : void {
+            if ($class !== ClassNotInPhar::class) {
+                return;
+            }
+
+            include_once __DIR__ . '/../../Fixture/ClassNotInPhar.php';
+        });
+
+        $a = new AutoloadableClassInPhar();
+        $this->assertInstanceOf(AutoloadableClassInPhar::class, $a);
+    }
+
+    /**
+     * @runInSeparateProcess
+     */
+    public function testPharDoesNotWork() : void
```

Comment:
Wording here needs adjustment - is it verifying that the source is not inside the phar?

Reply:
Renamed.